pkg/kube/util.go

```diff
@@ -83,10 +99,16 @@ func CreateClientset(kubeconfig, context string, fns ...func(*rest.Config)) (*ku
 }
 
 // DefaultRestConfig returns the rest.Config for the given kube config file and context.
-func DefaultRestConfig(kubeconfig, configContext string, fns ...func(*rest.Config)) (*rest.Config, error) {
-	config, err := BuildClientConfig(kubeconfig, configContext)
-	if err != nil {
-		return nil, err
+func DefaultRestConfig(ns string, kubeconfig, configContext string, fns ...func(*rest.Config)) (*rest.Config, error) {
```

Comment:
Having DefaultRestConfig pull in cluster config, create a client, fetch a secret, then create a new client seems far outside the scope of this method. This is not clear at all from the call, and you are completely modifying the semantics of DefaultRestConfig. This impacts literally every part of the code. For example, istioctl will now attempt to get in cluster credentials and do multicluster behavior

Reply:
Agree, will move it outside DefaultRestConfig and guard with our `externalIstiod` flag once it landed.